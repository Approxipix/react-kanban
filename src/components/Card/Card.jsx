import React, { Component } from 'react';
import { connect } from "react-redux";
import { Draggable } from 'react-beautiful-dnd';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

const Container = styled.div`
  position: relative;
  padding: .5rem;
  border-radius: .2rem;
  background-color: #fff;
  box-shadow: 0 1px 0 rgba(9,30,66,.25);  
  transition: background-color .2s ease-in;
  &:hover {
    background-color: #f4f5f7;
  }  
  margin-bottom: .5rem;
`;

const Title = styled.h3`
  font-size: .875rem;
  color: #40424b;
  font-weight: normal;
`;

const CardLabelList = styled.div`
  display: flex;
  margin-bottom: .3rem;
`;

const CardLabelListItem = styled.div`
  height: .5rem;
  width: 2.5rem;
  border-radius: 1rem;
  margin-right: .2rem;
  background-color: ${props => props.value};
`;

class CardLabels extends Component {
  render() {
    const { cardLabels } = this.props;
    if (cardLabels.length === 0) return null;
    return  (
      <CardLabelList>
        {cardLabels.map(((label, index) => (
          <CardLabelListItem key={index} value={label.color}/>
        )))}
      </CardLabelList>
    )
  }
}

class Card extends Component {
  render() {
    const { currentBoardID, card, index, isDraggingOver} = this.props;
    return (
      <Draggable
        index={index}
        draggableId={`${card._cardId}`}
      >
        {(provided, snapshot) => (
          <>
            <Container
              {...provided.draggableProps}
              {...provided.dragHandleProps}
              ref={provided.innerRef}
              isDragging={snapshot.isDragging}
            >
              <NavLink to={`/b/${currentBoardID}/c/${card._cardId}`}>
                <CardLabels cardLabels={card.cardLabels} />
                <Title>{card.title}</Title>
              </NavLink>
            </Container>
            {isDraggingOver && provided.placeholder}
          </>
        )}
      </Draggable>
    )
  }
}

function mapStateToProps(state, ownProps) {
  return {
    currentBoardID: state.rootReducer.currentBoardID,
    card: state.cardReducer[ownProps.cardId],
  }
}

export default connect(mapStateToProps)(Card);
