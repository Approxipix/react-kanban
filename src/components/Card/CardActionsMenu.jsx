import React, { Component } from 'react';
import styled from 'styled-components';
import { connect } from "react-redux";
import { bindActionCreators } from 'redux';
import { addLabelToCard } from '../../redux/cardReducer/actions';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import LabelTooltip from '../Tooltip/LabelTooltip';
import CheckListTooltip from '../Tooltip/CheckListTooltip';

const Wrapper = styled.div`
  position: relative;
  margin-bottom: 1rem;
  padding-left: 1.8rem;
`;

const ButtonWrapper = styled.div`
  position: relative;
  &:not(:last-child) {
    margin-bottom: .5rem;
  }
`;


const Title = styled.h4`
  margin-bottom: .5rem;
  font-size: .75rem;
  text-transform: uppercase;
`;

const Actions = styled.div`
  display: flex;
  flex-direction: column;
`;

const Button = styled.button`
  display: flex;
  align-items: center;
  width: 100%;
  padding: .5rem .75rem;
  font-size: .875rem;
  font-weight: 600;
  border-radius: .2rem;
  text-align: left;
  background-color: #ebecf0;
  box-shadow: 0 1px 0 0 rgba(9,30,66,.13);
  cursor: pointer;
`;

const ButtonIcon = styled.div`
  margin-right: .5rem;
`;

class CardActionsMenu extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLabelTooltipOpened: false,
      isCheckListTooltipOpened: false,
    }
  }

  toggleLabelTooltip = () => {
    this.setState({
      isLabelTooltipOpened: !this.state.isLabelTooltipOpened
    })
  };

  toggleCheckListTooltip = () => {
    this.setState({
      isCheckListTooltipOpened: !this.state.isCheckListTooltipOpened
    })
  };

  handelAddLabel = (label) => {
    this.props.actions.addLabelToCard({
      cardId: this.props.card._cardId,
      cardLabel: label
    })
  };

  render() {
    const { isLabelTooltipOpened, isCheckListTooltipOpened } = this.state;
    const { card } = this.props;
    return (
      <Wrapper>
        <Title>
          Add to cards
        </Title>
        <Actions>
          <ButtonWrapper>
            <Button onClick={() => this.toggleLabelTooltip()}>
              <ButtonIcon>
                <FontAwesomeIcon icon="tag" />
              </ButtonIcon>
              Label
            </Button>
            {isLabelTooltipOpened && (
              <LabelTooltip
                cardId={card._cardId}
                cardLabels={card.cardLabels}
                handelAddLabel={this.handelAddLabel}
                toggleLabelTooltip={this.toggleLabelTooltip}
              />
            )}
          </ButtonWrapper>
          <ButtonWrapper>
            <Button onClick={() => this.toggleCheckListTooltip()}>
              <ButtonIcon>
                <FontAwesomeIcon icon="check-square" />
              </ButtonIcon>
              CheckList
            </Button>
            {isCheckListTooltipOpened && (
              <CheckListTooltip
                cardId={card._cardId}
                toggleCheckListTooltip={this.toggleCheckListTooltip}
              />
            )}
          </ButtonWrapper>
        </Actions>
      </Wrapper>
    )
  }
}

function mapStateToProps(state) {
  return {

  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      addLabelToCard: addLabelToCard,
    }, dispatch)
  };
}


export default connect(mapStateToProps, mapDispatchToProps)(CardActionsMenu);
