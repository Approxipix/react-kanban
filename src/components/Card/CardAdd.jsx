import React, { Component } from "react";
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import { bindActionCreators } from 'redux';
import { addCard } from '../../redux/cardReducer/actions/actions';
import { TextArea, Actions, SubmitButton, CancelButton } from '../BaseComponent';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import ClickOutside from "../ClickOutside/ClickOutside";
import styled from 'styled-components';

const Wrapper = styled.div`
  padding: .5rem;
`;

const AddButton = styled.button`
  text-align: left;
  width: 100%;
  padding: .5rem .75rem;
  font-size: .875rem; 
  color: #6b778c;
  transition: all .2s ease-in;
  cursor: pointer;
  &:hover {
    color: #172b4d;
    background-color: rgba(9,30,66,.13);
  }
`;
AddButton.displayName = 'AddButton';

class CardAdd extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpened: false,
      title: '',
    };
    this.textarea = React.createRef();
  }

  toggleOpened = () => {
    this.setState({
      isOpened: !this.state.isOpened,
      title: '',
    })
  };

  handleKeyDown = (e) => {
    if (e.keyCode === 27) {
      this.toggleOpened();
    }
  };

  handleChange = (e) => {
    this.setState({
      title: e.target.value,
    });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { title } = this.state;
    if (!title)  return;
    this.props.actions.addCard({
      listId: this.props.listId,
      cardTitle: title,
      newCardId: Date.now().toString(),
    });
    this.setState({ title: '' }, () => {
      if (!this.textarea.focus) return;
      this.textarea.focus()
    })
  };

  render = () => {
    const { isOpened, title } = this.state;

    return isOpened ? (
      <ClickOutside action={'mousedown'} toggleOpened={this.toggleOpened}>
        <Wrapper>
          <form onSubmit={this.handleSubmit}>
            <TextArea
              type="text"
              name="title"
              value={title}
              size=".875rem;"
              padding=".5rem"
              rows={3}
              placeholder="Enter a card title..."
              ref={(e) => { this.textarea = e }}
              onKeyDown={this.handleKeyDown}
              onChange={(e) => this.handleChange(e)}
              spellCheck={false}
              autoFocus
            />
            <Actions>
              <SubmitButton type="submit" disabled={!title}>
                Create card
              </SubmitButton>
              <CancelButton onClick={() => this.toggleOpened()}>
                <FontAwesomeIcon icon="times"/>
              </CancelButton>
            </Actions>
          </form>
        </Wrapper>
      </ClickOutside>
    ) : (
      <AddButton onClick={() => this.toggleOpened()}>
        + Add another card
      </AddButton>
    );
  };
}

CardAdd.propTypes = {
  listId: PropTypes.string.isRequired,
};

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      addCard: addCard,
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(CardAdd);
