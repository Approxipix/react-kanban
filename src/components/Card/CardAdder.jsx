import React, { Component } from "react";
import { connect } from "react-redux";
import styled from 'styled-components'
import { bindActionCreators } from 'redux';
import { addCard } from '../../redux/rootReducer/actions';
import { Input, SubmitButton, CancelButton } from '../BaseComponent';
import ClickOutside from "../ClickOutside";

const Wrapper = styled.div`
  position: relative;
  padding: .75rem .5rem;
  background-color: #ddd;
  border-radius: .4rem
`;

const Actions = styled.div`
  display: flex;
  align-items: center;
`;

const Form = styled.form``;

const AddButton = styled.button`
  text-align: left;
  width: 100%;
  padding: .75rem;
  font-size: 1rem; 
  color: #444;
  font-weight: 600;
  transition: background-color .1s;
  cursor: pointer;
  &:hover {
    background-color: #ccc;
  }
`;

class CardAdder extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpened: false,
      title: '',
    };
  }

  toggleOpened = () => {
    this.setState({
      isOpened: !this.state.isOpened,
      title: '',
    })
  };

  handleChange = (key, value) => {
    this.setState({
      ...this.state,
      [key]: value,
    });
  };

  handleSubmit = event => {
    event.preventDefault();
    const { title } = this.state;
    if (!title)  return;
    this.props.actions.addCard({
      listId: this.props.listId,
      cardTitle: title,
    });
    this.toggleOpened()
  };

  render = () => {
    const { isOpened, title } = this.state;
    return isOpened ? (
      <ClickOutside toggleOpened={this.toggleOpened}>
        <Wrapper>
          <Form id="card-add-form" onSubmit={this.handleSubmit}>
            <Input
              autoFocus
              type="text"
              placeholder="Enter a card title..."
              value={title}
              onChange={(e) => this.handleChange('title', e.target.value)}
              spellCheck={false}
            />
            <Actions>
              <SubmitButton type="submit" disabled={!title}>
                Create card
              </SubmitButton>
              or
              <CancelButton onClick={() => this.toggleOpened()}>
                cancel
              </CancelButton>
            </Actions>
          </Form>
        </Wrapper>
      </ClickOutside>
    ) : (
      <AddButton onClick={() => this.toggleOpened()}>
        + Add a card
      </AddButton>
    );
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      addCard: addCard,
    }, dispatch)
  };
}


export default connect(null, mapDispatchToProps)(CardAdder);
