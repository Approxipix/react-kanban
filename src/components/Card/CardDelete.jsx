import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { deleteCard } from '../../redux/cardReducer/actions/actions';
import { Text, SubmitButton } from '../BaseComponent';
import { history } from "../../redux/store";

class CardDelete extends Component {
  deleteList = () => {
    const { listId, cardId, currentBoardID } = this.props;
    this.props.actions.deleteCard({
      listId: listId,
      cardId: cardId,
    });
    history.push(`/b/${currentBoardID}`)
  };

  render() {
    return (
      <>
        <Text>
          All actions will be removed from the activity feed and you wonâ€™t be able to re-open the card.
          There is no undo.
        </Text>
        <SubmitButton color={'danger'} onClick={() => this.deleteList()}>
          Delete card
        </SubmitButton>
      </>
    );
  }
}

function mapStateToProps(state) {
  return {
    currentBoardID: state.rootReducer.currentBoardID,
  }
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      deleteCard: deleteCard,
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(CardDelete);
