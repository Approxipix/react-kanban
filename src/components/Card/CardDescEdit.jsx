import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { editCardDescription } from '../../redux/cardReducer/actions';
import styled from 'styled-components'
import { TextArea, SubmitButton, CancelButton } from '../BaseComponent';
import ClickOutside from '../ClickOutside';

const Form = styled.form`
  width: 90%;
`;

const Actions = styled.div`
  display: flex;
  align-items: center;
`;

class CardDescEdit extends Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.card.description || '',
    };
  }

  handleChange = (key, value) => {
    this.setState({
      ...this.state,
      [key]: value,
    });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { description } = this.state;
    const { cardIndex, listIndex } = this.props;
    if (!description)  return;
    this.props.actions.editCardDescription({
      cardDescription: description,
      cardIndex: cardIndex,
      listIndex: listIndex,
    });
    this.props.toggleIsDescEditing()
  };

  render() {
    const { description } = this.state;
    return (
      <ClickOutside toggleOpened={this.props.toggleIsDescEditing}>
        <Form id="card-edit-desc-form">
          <TextArea
            autoFocus
            type="text"
            rows={5}
            placeholder="Edit card description"
            value={description}
            onChange={(e) => this.handleChange('description', e.target.value)}
            spellCheck={false}
          />
          <Actions>
            <SubmitButton onClick={(e) => this.handleSubmit(e)}  disabled={!description}>
              Save
            </SubmitButton>
            or
            <CancelButton onClick={() => this.props.toggleIsDescEditing()}>
              cancel
            </CancelButton>
          </Actions>
        </Form>
      </ClickOutside>
    );
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      editCardDescription: editCardDescription,
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(CardDescEdit);
