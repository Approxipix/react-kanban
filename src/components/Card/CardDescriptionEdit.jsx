import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { editCardDesc } from '../../redux/cardReducer/actions';
import styled from 'styled-components'
import ClickOutside from '../ClickOutside';

const Form = styled.form`
  width: 90%;
`;

const Input = styled.input`
  width: 100%;
  padding: .3rem .5rem;
  font-size: 1.5rem;
  color: #40424b;
  font-weight: 600;
  border: none;
  box-shadow: none;
`;

class CardDescriptionEdit extends Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.cardDescription || '',
    };
  }

  handleChange = (key, value) => {
    this.setState({
      ...this.state,
      [key]: value,
    });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { description } = this.state;
    const { cardId } = this.props;
    this.props.actions.editCardDesc({
      cardDescription: description,
      cardId: cardId,
    });
    this.props.toggleIsDescEditing()
  };

  render() {
    const { description } = this.state;
    return (
      <ClickOutside toggleOpened={this.props.toggleIsDescEditing}>
        <Form onSubmit={this.handleSubmit} id="card-edit-title-form">
          <Input
            autoFocus
            type="text"
            placeholder="Edit card description"
            value={description}
            onChange={(e) => this.handleChange('description', e.target.value)}
            spellCheck={false}
          />
        </Form>
      </ClickOutside>
    );
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      editCardDesc: editCardDesc,
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(CardDescriptionEdit);
