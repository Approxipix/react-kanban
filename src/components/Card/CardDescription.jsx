import React, { Component } from 'react';
import styled from 'styled-components';
import CardDescriptionEdit from './CardDescriptionEdit';

const Desc = styled.h4`
  margin-bottom: .5rem;
  padding: .3rem .5rem;
  font-size: 1.5rem;
  font-weight: 600;
`;

const AddDescription = styled.div``;

class CardDescription extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isDescEditing: false,
    }
  }

  toggleIsDescEditing = () => {
    this.setState({
      isDescEditing: !this.state.isDescEditing
    })
  };

  render() {
    const { cardId, cardDescription } = this.props;
    const { isDescEditing } = this.state;

    if (isDescEditing) {
      return (
        <CardDescriptionEdit
          cardId={cardId}
          cardDescription={cardDescription}
          toggleIsDescEditing={this.toggleIsDescEditing}
        />
      )
    }
    if (!cardDescription) {
      return (
        <AddDescription onClick={() => this.toggleIsDescEditing()}>
          Add a more detailed descriptionâ€¦
        </AddDescription>
      )
    }
    return (
      <Desc onClick={() => this.toggleIsDescEditing()}>
        {cardDescription}
      </Desc>
    )
  }
}


export default CardDescription;
