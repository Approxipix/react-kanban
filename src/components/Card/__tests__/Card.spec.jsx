import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import Card, { CardBadges } from '../Card';

const mockStore = configureStore([]);
jest.mock('react-beautiful-dnd', () => ({
  Draggable: ({ children }) => children({
    draggableProps: {
      style: {},
    },
    innerRef: jest.fn(),
  }, {}),
}));

describe('<Card>', () => {
  let store;
  let props;
  let component;

  beforeEach(() => {
    store = mockStore({
      rootReducer: {
        currentBoardID: 'BoardID',
      },
      cardReducer: {
        CardID: {
          _cardId: 'CardID',
          title: 'Test Card',
          description: '',
          checkLists: [],
          cardLabels: [],
        }
      },
      checkListReducer: {}
    });
    props = {
      cardId: 'CardID',
    };

    const wrapper = shallow(<Card store={store} {...props} />);
    component = wrapper.find('Card').dive().dive();
  });

  it('should render with given state from Redux store', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});

describe('<CardBadges>', () => {
  it('should render description badge icon if description props is exist', () => {
    const props = {
      card: {
        description: 'Test Description',
        checkLists: [],
      }
    };
    const component = shallow(<CardBadges {...props} />).dive();

    expect(toJson(component)).toMatchSnapshot()
  });

  it('should render checkList badge icon if description props is exist', () => {
    const props = {
      card: {
        description: '',
        checkLists: ['CheckListID'],
      },
      checkLists: {
        CheckListID: {
          _checkListId: 'CheckListID',
          title: 'CheckList Title',
          items: [
            {status: true, description: '1'},
            {status: false, description: '2'},
          ]
        }
      }
    };
    const component = shallow(<CardBadges {...props} />).dive();

    expect(toJson(component)).toMatchSnapshot()
  });

  it('should sets the done prop as true if all checklist item done', () => {
    const props = {
      card: {
        description: '',
        checkLists: ['CheckListID'],
      },
      checkLists: {
        CheckListID: {
          _checkListId: 'CheckListID',
          title: 'CheckList Title',
          items: [
            {status: true, description: '1'},
            {status: true, description: '2'},
            {status: true, description: '3'},
          ]
        }
      }
    };
    const component = shallow(<CardBadges {...props} />).dive();

    expect(component.childAt(0).props().done).toEqual(true);
  });
});

