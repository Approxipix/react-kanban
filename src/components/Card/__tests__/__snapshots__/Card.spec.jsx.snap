// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card> should render with given state from Redux store 1`] = `
<Fragment>
  <styled.div
    style={
      Object {
        "background": undefined,
      }
    }
  >
    <styled.div>
      <NavLink
        to="/b/BoardID/c/CardID"
      >
        <styled.div>
          <Label
            cardLabels={Array []}
          />
        </styled.div>
        <styled.h3>
          Test Card
        </styled.h3>
        <CardBadges
          card={
            Object {
              "_cardId": "CardID",
              "cardLabels": Array [],
              "checkLists": Array [],
              "description": "",
              "title": "Test Card",
            }
          }
          checkLists={Object {}}
        />
      </NavLink>
    </styled.div>
  </styled.div>
</Fragment>
`;

exports[`<CardBadges> should render checkList badge icon if description props is exist 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
 
",
        ],
      },
      "displayName": "BadgeList",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": "div",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <styled.div
    done={false}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="check-square"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
     
    1
    /
    2
  </styled.div>
</StyledComponent>
`;

exports[`<CardBadges> should render description badge icon if description props is exist 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
 
",
        ],
      },
      "displayName": "BadgeList",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": "div",
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  <styled.div>
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="align-left"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </styled.div>
</StyledComponent>
`;
